contracts/farm/MasterBarista.sol

```diff
@@ -731,13 +816,37 @@ contract MasterBarista is IMasterBarista, OwnableUpgradeable, ReentrancyGuardUpg
   ) external override onlyStakeTokenCallerContract(_stakeToken) {
     uint256 multiplierBps = _getBonusMultiplierProportionBps(_lastRewardBlock, block.number);
     uint256 toBeLockedNum = _amount.mul(multiplierBps).mul(bonusLockUpBps);
+    _assignActiveToken();
 
     // mint & lock(if any) an extra reward
-    latte.mint(_to, _amount);
-    latte.lock(_to, toBeLockedNum.div(1e8));
-    latte.mint(devAddr, _amount.mul(devFeeBps).div(1e4));
-    latte.lock(devAddr, (toBeLockedNum.mul(devFeeBps)).div(1e12));
+    activeLatte.mint(_to, _amount);
+    activeLatte.lock(_to, toBeLockedNum.div(1e8));
+    activeLatte.mint(devAddr, _amount.mul(devFeeBps).div(1e4));
+    activeLatte.lock(devAddr, (toBeLockedNum.mul(devFeeBps)).div(1e12));
 
     emit MintExtraReward(_msgSender(), _stakeToken, _to, _amount);
   }
+
+  /// if reward token hasn't been set, set it as a latte
+  function _assignActiveToken() internal {
+    if (address(activeLatte) == address(0)) activeLatte = latte;
+    if (address(activeBean) == address(0)) activeBean = bean;
+  }
+
+  /// @notice migrate latteV1 -> latteV2 and beanV1 -> beanV2
+  function migrate(ILATTEV2 _latteV2, IBeanBag _beanV2) external onlyOwner {
+    massUpdatePools();
+    uint256 amount = latte.balanceOf(address(bean));
+
+    activeLatte = ILATTE(address(_latteV2));
+    activeBean = _beanV2;
+    latteV2 = _latteV2;
+    beanV2 = _beanV2;
+
+    bean.safeLatteTransfer(address(this), amount);
+    _latteV2.redeem(amount);
```

Comment:
need to `latte.safeApprove(address(latteV2), uint256(-1))` before `_latteV2.redeem(amount);`